src/pal/src/thread/process.cpp

```diff
@@ -1886,57 +1872,104 @@ CreateProcessModules(
+        // __LINKEDIT             0000000104476000-000000010447a000 [   16K] r--/rwx SM=COW  /Users/mikem/coreclr/bin/Product/OSx.x64.Debug/corerun
+        // Kernel Alloc Once      000000010447a000-000000010447b000 [    4K] rw-/rwx SM=PRV
+        // MALLOC (admin)         000000010447b000-000000010447c000 [    4K] r--/rwx SM=ZER
+        // ...
+        // MALLOC (admin)         00000001044ab000-00000001044ac000 [    4K] r--/rwx SM=PRV
+        // __TEXT                 00000001044ac000-0000000104c84000 [ 8032K] r-x/rwx SM=COW  /Users/mikem/coreclr/bin/Product/OSx.x64.Debug/libcoreclr.dylib
+        // __TEXT                 0000000104c84000-0000000104c85000 [    4K] rwx/rwx SM=PRV  /Users/mikem/coreclr/bin/Product/OSx.x64.Debug/libcoreclr.dylib
+        // __TEXT                 0000000104c85000-000000010513b000 [ 4824K] r-x/rwx SM=COW  /Users/mikem/coreclr/bin/Product/OSx.x64.Debug/libcoreclr.dylib
+        // __TEXT                 000000010513b000-000000010513c000 [    4K] rwx/rwx SM=PRV  /Users/mikem/coreclr/bin/Product/OSx.x64.Debug/libcoreclr.dylib
+        // __TEXT                 000000010513c000-000000010516f000 [  204K] r-x/rwx SM=COW  /Users/mikem/coreclr/bin/Product/OSx.x64.Debug/libcoreclr.dylib
+        // __DATA                 000000010516f000-00000001051ce000 [  380K] rw-/rwx SM=COW  /Users/mikem/coreclr/bin/Product/OSx.x64.Debug/libcoreclr.dylib
+        // __DATA                 00000001051ce000-00000001051fa000 [  176K] rw-/rwx SM=PRV  /Users/mikem/coreclr/bin/Product/OSx.x64.Debug/libcoreclr.dylib
+        // __LINKEDIT             00000001051fa000-0000000105bac000 [ 9928K] r--/rwx SM=COW  /Users/mikem/coreclr/bin/Product/OSx.x64.Debug/libcoreclr.dylib
+        // VM_ALLOCATE            0000000105bac000-0000000105bad000 [    4K] r--/rw- SM=SHM
+        // MALLOC (admin)         0000000105bad000-0000000105bae000 [    4K] r--/rwx SM=ZER
+        // MALLOC                 0000000105bae000-0000000105baf000 [    4K] rw-/rwx SM=ZER
+        char *line = NULL;
+        size_t lineLen = 0;
+        int count = 0;
+        ssize_t read;
+
+        char vmmapCommand[100];
+        int chars = snprintf(vmmapCommand, sizeof(vmmapCommand), "vmmap -interleaved %d", dwProcessId);
+        _ASSERTE(chars > 0 && chars <= sizeof(vmmapCommand));
+
+        FILE *vmmapFile = popen(vmmapCommand, "r");
+        if (vmmapFile == NULL)
+        {
+            SetLastError(ERROR_INVALID_HANDLE);
+            return NULL;
+        }
+
+        // Reading maps file line by line
+        while ((read = getline(&line, &lineLen, vmmapFile)) != -1)
         {
-            if (inode != 0)
+            void *startAddress, *endAddress;
+            char moduleName[PATH_MAX];
+            int size;
+
+            if (sscanf(line, "__TEXT %p-%p [ %dK] %*[-/rwxsp] SM=%*[A-Z] %s\n", &startAddress, &endAddress, &size, moduleName) == 4)
```

Comment:
Its also worth noting that a hijacked vmmap could corrupt the stack here I think.